internal/pkg/iptablesutils/iptables.go

```diff
@@ -61,8 +61,20 @@ func DetectIPTablesMode(k0sBinPath string) (string, error) {
 		logrus.Info("kube-related iptables entries not found, go with iptables-legacy")
 		return ModeLegacy, nil
 	}
-	logrus.Info("kube-related iptables entries not found, go with iptables-nft")
-	return ModeNFT, nil
+
+	iptablesPath, err := exec.LookPath("iptables")
+	if err != nil {
+		return "", err
+	}
+	out, err := exec.Command(iptablesPath, "--version").CombinedOutput()
+	if err != nil {
+		return "", err
+	}
+	outStr := string(out)
+	if strings.Contains(outStr, "nf_tables") {
+		return ModeNFT, nil
+	}
+	return ModeLegacy, nil
```

Comment:
Do we want to use `ModeLegacy` here regardless of the kernel version?

Reply:
Yes. If `iptables --version` successfully executes, but does not return "nf_tables", it uses legacy mode.

I think we want use legacy mode if there are existing iptables on the host in legacy mode. Otherwise things will go wrong if user adds firewall rules after k0s has been started.